subsys/net/ip/net_shell.c

```diff
@@ -2698,6 +2698,104 @@ static int cmd_net_events(const struct shell *shell, size_t argc, char *argv[])
 	return 0;
 }
 
+#if defined(CONFIG_NET_DHCPV4)
+
+static struct net_if * cmd_net_dhcp_get_interface(const struct shell *shell, size_t argc, char * argv[])
+{
+
+	struct net_if *iface = NULL;
+
+	if (argc == 2) {
+		int index = atoi(argv[1]);
+		iface = net_if_get_by_index(index);
+	} else if (argc == 1) {
+		iface = net_if_get_default();
+	} else {
+		PR_ERROR("DHCPv4: invalid number of arguments.\n");
+		return NULL;
+	}
+
+	if (iface == NULL)
+	{
+		PR_ERROR("DHCPv4: no interface found.\n");
+		return NULL;
```

Comment:
Just let teh function return `iface` directly if it's NULL, no need for another exit point.